### DISCUSSION TITLE
Distacco di emergenza in ordine

### DISCUSSION THREAD

**Discussion by MikZenn** [Home-Assistant-Pro-Team/Power-Control-HomeAssistant#8] [General]
Ciao.

è possibile inserire un opzione per cui anche quando si raggiunge la soglia di emergzan si stacca in ordine invece che l'ultimo disposiyivo collegato se negli ultimi 30 sec?
Avendo una lista di priorità vorrei fosse sempre quella a comandare.
Grazie e ottimo progetto.

**Comment by marco-hacs** [comment]
Volendo puoi creare un nuovo inpupt_boolean e aggiungerlo come condizione alla riga 231. 

Oppure in  alternativa cancellare la sequenza if, then, e lasciare solo la sequenza else controllando la giusta indentazione.

**Comment by ebaratto2** [comment]
anche io ho lo stesso problema e mi piacerebbe una tua modifica ufficiale dello script (così ad ogni aggiornamento non devo modificare a mano il funzionamento del tuo automatismo).
Faccio un esempio su una fatto che mi è capitato: 
accendo la lavatrice che parte con le sue attività preliminari (fase 1 del lavaggio). Tali attività durano alcuni minuti ed hanno un basso impatto nei KW consumati
Terminata questa fase la lavatrice inizia a scaldare l'acqua ed a consumare molti KW (fase 2 del lavaggio). 
Se durante la fase 2 supero la soglia di emergenza il tuo automatismo mi spegnerà la lavatrice e non la riaccenderà mai più.
Questo comporta che il tuo ottimo sistema di controllo carichi non è più "automatico" ma "manuale".
Ieri sono uscito per un paio d'ore mentre la lavatrice era in fase 1 del lavaggio. Al rientro a casa ho trovato la lavatrice senza alimentazione elettrica e con il bucato da lavare. 
In buona sostanza ciò che viene spento deve poi essere riacceso a meno che non sia stato messo all'interno delle esclusioni.
Ringrazio in anticipo

**Comment by marco-hacs** [comment]
Sto facendo delle modifiche dove appunto verrà cancellata o resa indipendente questa funzione... E come suggerito in un altro commento prima di eseguire la riaccensione del dispositivo controllo quanto consumava.

**Reply by marco-hacs** [reply]
La funzione è stata rimossa potete aggiornare
![image](https://github.com/Home-Assistant-Pro-Team/Power-Control-HomeAssistant/assets/62516592/b2c81541-1faa-4379-a9bf-d32e26a34a46)

**Reply by MikZenn** [reply]
Bellissime modifiche. Soprattutto quella della lista consumi individuali.

Non ho capito bene che intendi con il punto 3) Aggiunto il controllo consumo per singolo dispositivo prima di ripristinare carico.

Io vedo solo quello del punto 6).

Sempre complimenti e grazie

**Reply by marco-hacs** [reply]
Il punto 3... prima i dispositivi venivano riavviati nell'ordine della lista ogni 20 secondi dal momento in cui il consumo tornava regolare. Adesso se il dispositivo da accendere ha il sensore Power viene salvato il consumo al momento del distacco e per il ripristino verificato che la somma del consumo del dispositivo con il consumo istantaneo di casa non superi la soglia di distacco. In questo modo si evitano loop di spegni accendi inutili. Mentre se il dispositivo non ha il sensore Power il funzionamento rimane invariato a prima

**Comment by MikZenn** [comment]
Wow, fighissimo. Grande :)

Il giorno gio 28 mar 2024 alle ore 13:59 Marco ***@***.***>
ha scritto:
